roles/ntp-server/tasks/main.yml

```diff
@@ -0,0 +1,119 @@
+---
+- name: Check if ntp package installed
+  command: rpm -q ntp
+  ignore_errors: true
+  register: ntp_installed
+
+- name: Check if chrony package installed
+  command: rpm -q chrony
+  ignore_errors: true
+  register: chrony_installed
+
+# Use NTP if neither time service is installed
+- set_fact:
+    use_ntp: true
+    use_chrony: false
+  when:
+    - ntp_installed.rc != 0
+    - chrony_installed.rc != 0
+
+# Use NTP if it's installed and Chrony isn't
+- set_fact:
+    use_ntp: true
+    use_chrony: false
+  when:
+    - ntp_installed.rc == 0
+    - chrony_installed.rc != 0
+
+# Use Chrony if it's installed and NTP isn't
+- set_fact:
+    use_ntp: false
+    use_chrony: true
+  when:
+    - ntp_installed.rc != 0
+    - chrony_installed.rc == 0
+
+# It's unlikely we have four baremetal hosts doing nothing but serving as NTP servers.
+# Thus, we shouldn't go uninstalling anything since either package could be a dependency
+# of an already running service.
+- fail:
+    msg: "Both NTP and Chrony are installed.  Check dependencies before removing either package and proceeding."
+  when:
+    - ntp_installed.rc == 0
+    - chrony_installed.rc == 0
+
+- name: Install and update ntp package
+  yum:
+    name: ntp
+    state: latest
+  when: use_ntp == true
+
+- name: Install and update chrony package
+  yum:
+    name: chrony
+    state: latest
+  when: use_chrony == true
+
+- name: Write NTP config file
+  template:
+    src: ntp.conf.j2
+    dest: /etc/ntp.conf
+  register: conf_written
+  when: use_ntp == true
+
+- name: Write chronyd config file
```

Comment:
Does this write the file for all rpm systems when chronyd is enabled? 

@ktdreyer do you have any comments here? ceph-ansible uses ntp i guess which we can make as standard here as well.

Reply:
@vasukulkarni this is only for NTP *servers*.   You don't need to worry about testnodes.

Reply:
sorry forgot, it is doing much more than testnodes 👍